### PR TITLE
contracts-bedrock: migrate to new storage snapshot

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Delete the old storage layout now that https://github.com/ethereum-optimism/optimism/pull/8243
has been merged which adds another snapshots concept of both the abi and
the storage layout.

The old method of keeping the storage layout lockfile around is now
deprecated. Once we delete it, we will no longer be able to track
changes from old versions of the contracts as easily. We can reintroduce
the shell script to a future commit if necessary.

## Summary by CodeRabbit

- **Chores**
  - Streamlined CI pipeline by consolidating snapshot-related tasks.
  - Improved clarity in CI failure messages related to snapshot processes.

### CODE REVIEW

**Review by mslipper — APPROVED** [review]
infra changes lgtm

File: .circleci/config.yml [hunk 1]
```diff
@@ -420,22 +420,16 @@ jobs:
             FOUNDRY_PROFILE: ci
           working_directory: packages/contracts-bedrock
           no_output_timeout: 15m
-      - run:
-          name: storage snapshot
-          command: |
-            pnpm storage-snapshot
-            git diff --exit-code .storage-layout || echo "export STORAGE_SNAPSHOT_STATUS=1" >> "$BASH_ENV"
```
Comment (clabby) [inline .circleci/config.yml:430]: Looks like there's still a check for this env var below that needs to be removed

File: .circleci/config.yml [hunk 2]
```diff
@@ -420,22 +420,16 @@ jobs:
...
             git diff --exit-code ./invariant-docs/*.md || echo "export INVARIANT_DOCS_STATUS=1" >> "$BASH_ENV"
           working_directory: packages/contracts-bedrock
       - run:
-          name: abi snapshot
+          name: snapshots
           command: |
-            pnpm abi-snapshot
+            pnpm snapshots
             if [ "$(git diff --exit-code snapshots)" ]; then
               [ -z "$(git ls-files --others --exclude-standard snapshots)" ] || echo "export ABI_SNAPSHOT_STATUS=1" >> "$BASH_ENV"
```
Comment (clabby) [inline .circleci/config.yml]: nit: env var is named for ABI snapshots, can we rename it to something more general?

File: .circleci/config.yml (hunk 1 above)
Comment (tynes) [reply .circleci/config.yml:430]: good catch, this has been addressed

File: .circleci/config.yml (hunk 2 above)
Comment (tynes) [reply .circleci/config.yml]: also good catch, addressed this

**Review by refcell — APPROVED** [review]

**Pull request merged**